Update publication info graph with publication date using PubMed Id

ImranAsif48 opened the pull request:
The reason for this Pull request is that to find the timeline of discourse. I updated the code that fetched the PubMed ID's publication date using the PubMed API (In PubMed Java class). After fetching the publication date, I put it in the publication information graph with pav:authoredOn predicate. I also updated the dcterms:created to pav:createdOn because pav:createdOn is the date of creating the digital artefact or resource representation.

ImranAsif48 committed:
Testing

ImranAsif48 committed:
The reason for this Pull request is that to find the timeline of
discourse. I updated the code that fetched the PubMed ID's publication
date using the PubMed API (In PubMed Java class). After fetching the
publication date, I put it in the publication information graph with
pav:authoredOn predicate. I also updated the dcterms:created to
pav:createdOn because pav:createdOn is the date of creating the digital
artefact or resource representation.

egonw commented on src/main/java/nl/unimaas/bigcat/wikipathways/nanopubs/PubMed.java:
```diff
@@ -0,0 +1,96 @@
+			    	
+			    	conn.disconnect();
+			    	
+			    	if (!((String)pubmedJson.get("epubdate")).equals(""))
```
When I run the code, I get this exception:

```
Exception in thread "main" java.lang.NullPointerException
        at nl.unimaas.bigcat.wikipathways.nanopubs.PubMed.getPublicationDate(PubMed.java:48)
        at nl.unimaas.bigcat.wikipathways.nanopubs.NanoPubs.updateNanopublication(NanoPubs.java:147)
        at nl.unimaas.bigcat.wikipathways.nanopubs.NanoPubs.createNanoPublications(NanoPubs.java:103)
        at nl.unimaas.bigcat.wikipathways.nanopubs.ComplexesPubs.main(ComplexesPubs.java:32)
```

ImranAsif48 replied:
I hope you will not get this exception now.

egonw commented on src/main/java/nl/unimaas/bigcat/wikipathways/nanopubs/PubMed.java:
```diff
@@ -0,0 +1,96 @@
+package nl.unimaas.bigcat.wikipathways.nanopubs;
```
Please add a copyright/license header here like in https://github.com/wikipathways/nanopublications/blob/master/src/main/java/nl/unimaas/bigcat/wikipathways/nanopubs/ComplexesPubs.java

egonw commented on src/main/java/nl/unimaas/bigcat/wikipathways/nanopubs/PubMed.java:
```diff
@@ -0,0 +1,96 @@
+			    String postUrl = "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=pubmed&id="+pubmedId+"&retmode=json"; //post url
+			    //URLConnection conn = null;
+			    
+//	    		URL url = new URL(postUrl);
```
please remove outcommented code

ImranAsif48 replied:
Removed the out-commented code

egonw commented on src/main/java/nl/unimaas/bigcat/wikipathways/nanopubs/PubMed.java:
```diff
@@ -0,0 +1,96 @@
+	   public static Date getPublicationDate(String pubmedId) throws IOException, ParseException {
+		   if(isNumeric(pubmedId))
+		   {
+			    String postUrl = "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=pubmed&id="+pubmedId+"&retmode=json"; //post url
```
that is going to be a lot of webservice calls :(

ImranAsif48 replied:
I resolved the issue by creating the pubmed json data file

egonw reviewed (changes requested):
Please have a look at the exception and other comments.

Regarding the webservice calls, this suggests we really want the publication data as part of the RDF.

ImranAsif48 committed:
The pubmed publication date will be fetched from the json data file.
This file will overcome the lot of webservice calls. The name of the
file is pubmed_data.json which contains the article information such as
title, authors, and publication date.

ImranAsif48 reviewed (commented):
I resolved the issue by creating the pubmed json data file and uploaded it here. The issue was to fetch the pubmed API that gave the 500 HTTP response error. The pubmed publication date will be fetched from the json data file.This file will overcome the lot of webservice calls. The name of the file is pubmed_data.json which contains the article information such as title, authors, and publication date.
